pkg/rancher-desktop/main/__tests__/deploymentProfiles.spec.ts

```diff
@@ -5,118 +5,94 @@ import os from 'os';
+import {
+  readDeploymentProfiles,
+  lockableDefaultSettings, validateDeploymentProfile,
+} from '@pkg/main/deploymentProfiles';
 import { spawnFile } from '@pkg/utils/childProcess';
 import Logging from '@pkg/utils/logging';
 import { RecursivePartial } from '@pkg/utils/typeUtils';
 
-// for some reason `import nativeReg...` => undefined when run in jest on Windows
-// import nativeReg from 'native-reg';
-const nativeReg = require('native-reg');
-
 const console = Logging.deploymentProfile;
 
-// Note that we can't modify the HKLM hive without admin privileges,
-// so this whole test will just work with the user's HKCU hive.
-const REG_PATH_START = ['SOFTWARE', 'Rancher Desktop'];
-const FULL_REG_PATH_START = ['HKEY_CURRENT_USER'].concat(REG_PATH_START);
-const REGISTRY_PATH_PROFILE = REG_PATH_START.concat('TestProfile');
-
-const NON_PROFILE_PATH = FULL_REG_PATH_START.join('\\');
-const FULL_PROFILE_PATH = FULL_REG_PATH_START.concat('TestProfile').join('\\');
-
-const describeWindows = process.platform === 'win32' ? describe : describe.skip;
-
-let testDir = '';
-let regFilePath = '';
-
-async function clearRegistry() {
-  try {
-    await spawnFile('reg', ['DELETE', `HKCU\\${ REGISTRY_PATH_PROFILE.join('\\') }`, '/f']);
-  } catch {
-    // Ignore any errors
-  }
-}
-
-async function installInRegistry(regFileContents: string) {
-  await fs.promises.writeFile(regFilePath, regFileContents, { encoding: 'ascii' });
-  try {
-    await spawnFile('reg', ['IMPORT', regFilePath]);
-  } catch (ex: any) {
-    // Use expect to display the error message
-    expect(ex).toBeNull();
-    throw ex;
-  }
-}
-
-// Registry multi-stringSZ settings in a reg file are hard to read, so expand them here.
-// e.g.=> ["abc", "def"] would be ucs-2-encoded as '61,00,62,00,63,00,00,00,64,00,65,00,66,00,00,00,00,00'
-// where null dwords (so two 00 bytes) separate each pair of words and
-// two null dwords ("00 00 00 00") indicate the end of the list
-function stringToMultiStringHexBytes(s: string[]): string {
-  const hexBytes = Buffer.from(s.join('\x00'), 'ucs2')
-    .toString('hex')
-    .split(/(..)/)
-    .filter(x => x)
-    .join(',');
-
-  return `${ hexBytes },00,00,00,00`;
-}
-
-// We *could* write a routine that converts json to reg files, but that's not the point of this test.
-// Better to just hard-wire a few regfiles here.
-
-const defaultsUserRegFile = `Windows Registry Editor Version 5.00
+const [describeWindows, describeNotWindows] = process.platform === 'win32' ? [describe, describe.skip] : [describe.skip, describe];
+
+describe('deployment profiles', () => {
+  describeWindows('windows deployment profiles', () => {
+    // for some reason `import nativeReg...` => undefined when run in jest on Windows
+    // import nativeReg from 'native-reg';
+    const nativeReg = require('native-reg');
+
+    /* Mock console.error() to capture error messages. */
+    let consoleMock: jest.SpyInstance<void, [message?: any, ...optionalArgs: any[]]>;
```

Comment:
This could have been deleted once I built up a list of errors to throw in an exception. Yet again, I forgot to delete test-specific code that dealt with a deleted situation.